api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -223,6 +225,10 @@ else if (!obs.hasGroupMembers()) {
 				errors.rejectValue("concept", "must have a datatype");
 			}
 		}
+
+		if (!(obs.getPerson() == null && obs.getEncounter() ==null) && !Objects.equals(obs.getPerson(), obs.getEncounter().getPatient().getPerson())) {
```

Comment:
`Patient` is a subclass of `Person` so you don't need to call `getPerson()` on the Patient object, `encounter.getEncounter().getPatient()` should be enough.